src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java

```diff
@@ -47,49 +48,48 @@ public LetterService(LetterRepository letterRepository, Zipper zipper, ObjectMap
 
     @Transactional
     public UUID send(LetterRequest letter, String serviceName) {
-        Asserts.notEmpty(serviceName, "serviceName");
-
-        final String messageId = generateChecksum(letter);
-
-        log.info("Generated message: id = {}", messageId);
-
-        Optional<Letter> duplicateLetter = letterRepository.findByMessageIdAndStatusOrderByCreatedAtDesc(
-            messageId,
-            Created
+        return send(
+            new LetterInfo(generateChecksum(letter), serviceName, letter.type, letter.additionalData),
+            () -> pdfCreator.createFromTemplates(letter.documents)
         );
-
-        return duplicateLetter
-            .map(l -> {
-                UUID id = l.getId();
-                log.info("Same message found already created. Returning letter id {} instead", id);
-                return id;
-            })
-            .orElseGet(() -> saveNewLetterAndReturnId(letter, messageId, serviceName));
     }
 
     @Transactional
     public UUID send(LetterWithPdfsRequest letter, String serviceName) {
         throw new NotImplementedException();
```

Comment:
this will simply be changed to
```
return send(
    new LetterInfo(generateChecksum(letter), serviceName, letter.type, letter.additionalData),
    () -> pdfCreator.createFromBase64Pdfs(letter.documents)
);
```

Reply:
Looks pretty much the same. Can review `LetterInfo` usage. But I suggest for later once current changes get into scope to avoid premature refactor